hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -127,12 +128,59 @@ public static GenericRecord bytesToAvro(byte[] bytes, Schema schema) throws IOEx
    * Convert serialized bytes back into avro record.
    */
   public static GenericRecord bytesToAvro(byte[] bytes, Schema writerSchema, Schema readerSchema) throws IOException {
```

Comment:
How many places is this method being used ? Is adding namespace to the reader schema backwards compatible for data/schema written without the namespace ?

Reply:
@n3nash  Yes, it is very valid point. Adding namespace to reader schema might make it incompatible with data already written without namespace. 

I am thinking we should probably remove the namespace in the writerSchema fixed fields instead of adding namespace to the reader schema fixed fields. Since we are not altering reader schema in this approach, there shouldn't be incompatibility issue with data already written.

If you also agree with this approach, i will update `AvroConversionHelper.createConverterToAvro` method to remove namespace in writer schema fixed fields and revoke the changes made in `HoodieAvroUtils`

Reply:
@sathyaprakashg But like you mentioned, if you don't add namespace, you see the type mismatch exception, right ?

Reply:
@n3nash This is the execption we get when we don't add namespace to reader schema. `org.apache.avro.AvroTypeException: Found hoodie.source.hoodie_source.height.fixed, expecting fixed`. 

This exception is raised because we have namespace in writer schema and not in reader schema. As per my proposed approach above, if we remove the namespace from writer schema, then we wouldn't get type exception, because now namespace does not exist in fixed fields for both writer and reader schema